cmd/argocd/commands/app.go

```diff
@@ -1695,33 +1743,36 @@ func waitOnApplicationStatus(acdClient argocdclient.Client, appName string, time
 		var selectedResourcesAreReady bool
 
 		// If selected resources are included, wait only on those resources, otherwise wait on the application as a whole.
-		if len(selectedResources) > 0 {
+		if len(opts.selectedResources) > 0 {
 			selectedResourcesAreReady = true
-			for _, state := range getResourceStates(app, selectedResources) {
-				resourceIsReady := checkResourceStatus(watchSync, watchHealth, watchOperation, watchSuspended, state.Health, state.Status, appEvent.Application.Operation)
+			for _, state := range getResourceStates(app, opts.selectedResources) {
+				resourceIsReady := checkResourceStatus(opts.watch, state.Health, state.Status, appEvent.Application.Operation)
 				if !resourceIsReady {
 					selectedResourcesAreReady = false
 					break
 				}
 			}
 		} else {
 			// Wait on the application as a whole
-			selectedResourcesAreReady = checkResourceStatus(watchSync, watchHealth, watchOperation, watchSuspended, string(app.Status.Health.Status), string(app.Status.Sync.Status), appEvent.Application.Operation)
+			selectedResourcesAreReady = checkResourceStatus(opts.watch, string(app.Status.Health.Status), string(app.Status.Sync.Status), appEvent.Application.Operation)
 		}
 
-		if selectedResourcesAreReady && (!operationInProgress || !watchOperation) {
+		if selectedResourcesAreReady && (!operationInProgress || !opts.watch.operation) && !opts.watch.delete {
 			app = printFinalStatus(app)
 			return app, nil
 		}
 
-		newStates := groupResourceStates(app, selectedResources)
+		newStates := groupResourceStates(app, opts.selectedResources)
 		for _, newState := range newStates {
 			var doPrint bool
 			stateKey := newState.Key()
 			if prevState, found := prevStates[stateKey]; found {
-				if watchHealth && prevState.Health != string(health.HealthStatusUnknown) && prevState.Health != string(health.HealthStatusDegraded) && newState.Health == string(health.HealthStatusDegraded) {
+				if opts.watch.health && prevState.Health != string(health.HealthStatusUnknown) && prevState.Health != string(health.HealthStatusDegraded) && newState.Health == string(health.HealthStatusDegraded) {
 					_ = printFinalStatus(app)
-					return nil, fmt.Errorf("application '%s' health state has transitioned from %s to %s", appName, prevState.Health, newState.Health)
+					return nil, fmt.Errorf("application '%s' health state has transitioned from %s to %s", opts.appName, prevState.Health, newState.Health)
+				}
+				if opts.watch.delete && newState.Health == string(health.HealthStatusMissing) {
+					newState.Message = strings.Replace(newState.Message, "created", "deleted", 1)
```

Comment:
Sometimes the wait command prints that the resource is deleted even though it is only being synced. In the below message, the deployment.apps/test is shown as deleted during sync

```shell
2022-03-03T14:18:38+05:30            Service            test                  test  OutOfSync  Missing              
2022-03-03T14:18:57+05:30         ServiceAccount        test                  test    Synced  Missing              
2022-03-03T14:18:57+05:30            Service        test                  test    Synced  Healthy              
2022-03-03T14:18:57+05:30         ServiceAccount        test                  test    Synced   Missing              serviceaccount/test created
2022-03-03T14:18:57+05:30            Service            test                  test    Synced   Healthy              service/test created
2022-03-03T14:18:57+05:30   apps  Deployment            test                  test  OutOfSync  Missing              deployment.apps/test deleted
2022-03-03T14:18:57+05:30   apps  Deployment        test                  test    Synced  Progressing              deployment.apps/test created
2022-03-03T14:19:08+05:30   apps  Deployment        test                  test    Synced  Healthy              deployment.apps/test created
```
Steps to reproduce:
1. Create an app with manual sync
2. argocd app wait test --delete
3. Sync the application